texar/torch/modules/networks/conv_networks.py

```diff
@@ -517,13 +572,46 @@ def forward(self,  # type: ignore
                 layers.
             dtype (optional): Type of the inputs. If not provided,
                 infers from inputs automatically.
+            data_format (optional): Data type of the input tensor. If
+                ``channels_last``, the last dimension will be treated as channel
+                dimension. If ``channels_first``, first dimension will be
+                treated as channel dimension. Defaults to None.
+                If None, the value will be picked from hyperparameters.
         Returns:
             The output of the final layer.
         """
-        if sequence_length is not None:
-            input = mask_sequences(input, sequence_length,
-                                   dtype=dtype, time_major=False)
-        return super().forward(input)
+        if data_format is None:
+            data_format = self.hparams["data_format"]
+
+        if data_format == "channels_first":
+            # masking requires channels in last dimension
+            input = input.permute(0, 2, 1)
+
+            if sequence_length is not None:
+                input = mask_sequences(input, sequence_length,
+                                       dtype=dtype, time_major=False)
+
+            # network is constructed for channel first tensors
+            input = input.permute(0, 2, 1)
+
+            output = super().forward(input)
+
+        elif data_format == "channels_last":
+            if sequence_length is not None:
+                input = mask_sequences(input, sequence_length,
+                                       dtype=dtype, time_major=False)
+
+            input = input.permute(0, 2, 1)
+
+            output = super().forward(input)
+
+            # transpose only when no dense layers
+            if self._hparams.num_dense_layers <= 0:
+                output = output.permute(0, 2, 1)
```

Comment:
What if `Conv1DClassifier` append a `Flatten` layer and a `Linear` layer for the encoder? In this case the `output` may have a shape [batch_size, ...] and can't be permuted.

Reply:
No problem. We value the feedback. Good find! A more elegant check I think is `if output.dim() == 3`

Reply:
I think a more appropriate check would be `if output.dim() > 3`. Also, this can be rewritten as `output.transpose(1, 2)` which works for more than 3 dimensions.

Reply:
I think you meant `output.dim() >= 3`? However, since we use `Conv1D`, our tensors are 3D, we cannot perform Conv1D on higher than 3D tensors, right?

Reply:
Ah yes you're right. Sorry. Then let's add a check to ensure that the input does not contain more than 3 dimensions.

Reply:
Added a check in 373773c